client/resteasy-client/src/main/java/com/cerner/beadledom/client/resteasy/ExponentialBackoffHttpRequestRetryHandler.java

```diff
@@ -0,0 +1,49 @@
+package com.cerner.beadledom.client.resteasy;
+
+import java.io.IOException;
+import java.util.Random;
+import org.apache.http.impl.client.StandardHttpRequestRetryHandler;
+import org.apache.http.protocol.HttpContext;
+
+/**
+ * A {@link StandardHttpRequestRetryHandler} that adds an exponential back-off to retries.
+ *
+ * @author John Leacox
+ * @since 3.5
+ */
+class ExponentialBackoffHttpRequestRetryHandler extends StandardHttpRequestRetryHandler {
+  private static final int INITIAL_DELAY = 100;
+  private static final int BACKOFF_FACTOR = 2;
+  private static final Random random = new Random();
+
+  ExponentialBackoffHttpRequestRetryHandler(int retryCount, boolean requestSentRetryEnabled) {
+    super(retryCount, requestSentRetryEnabled);
+  }
+
+  @Override
+  public boolean retryRequest(IOException exception, int executionCount, HttpContext context) {
```

Comment:
Should we respect a `Retry-After` header if there is one on the request? 
https://tools.ietf.org/html/rfc7231#section-7.1.3

Reply:
I think handling of that would be in the Service Unavailable retry strategy. This retry handler is for IOExceptions, so there won't be any response or headers to look at.